app/services/katello/pulp/repository.rb

```diff
@@ -0,0 +1,21 @@
+module Katello
+  module Pulp
+    class Repository < ::Actions::Pulp::Abstract
+      attr_accessor :repo
+
+      def initialize(repo)
+        @repo = repo
+      end
+
+      def sync(overrides = {})
+        sync_options = {}
+        sync_options[:max_speed] = SETTINGS.dig(:katello, :pulp, :sync_KBlimit)
+        sync_options[:num_threads] = SETTINGS.dig(:katello, :pulp, :sync_threads)
+        sync_options[:feed] = overrides[:source_url] if overrides[:source_url]
+        sync_options[:validate] = !SETTINGS.dig(:katello, :pulp, :skip_checksum_validation)
+        sync_options.merge!(overrides[:options]) if overrides[:options]
+        [pulp_resources(overrides[:capsule_id]).repository.sync(@repo.pulp_id, override_config: sync_options.compact!)]
```

Comment:
pulp_resources here won't work, will it?

Reply:
We need a way to pass the capsule id to the abstract class..I saw something similar here : https://github.com/Katello/katello/blob/27d5de2493bfab921c3bf2060ea8f2fcfb364561/app/lib/actions/pulp/repository/refresh.rb#L11

Reply:
that works because its in an action (and the pulp_extensions method is defined in the abstract pulp action).  Some of what we do is going to depend on what smart proxy we are used, so i think that should be part of the initializer:  

`def initialize(repo, smart_proxy)` and then we can reference it in the same way we reference the repo